flink-connector/flink/src/main/java/org/apache/gravitino/flink/connector/catalog/BaseCatalog.java

```diff
@@ -521,12 +554,30 @@ protected CatalogBaseTable toFlinkTable(Table table) {
           .column(column.name(), column.nullable() ? flinkType.nullable() : flinkType.notNull())
           .withComment(column.comment());
     }
+    handleFlinkPrimaryKey(table, builder);
     Map<String, String> flinkTableProperties =
         propertiesConverter.toFlinkTableProperties(table.properties());
     List<String> partitionKeys = partitionConverter.toFlinkPartitionKeys(table.partitioning());
     return CatalogTable.of(builder.build(), table.comment(), partitionKeys, flinkTableProperties);
   }
 
+  private static void handleFlinkPrimaryKey(
```

Comment:
could you use `String getFlinkPrimaryKey(Table table)` to return the primary field name,  builder could use field name to build primary key.

Reply:
done